doc/common-illnesses.rst

```diff
@@ -0,0 +1,157 @@
+
+To provide code completions, help and real-time analysis in the Editor, Spyder uses the Python Language Server (PyLS), an implementation of the Language Server Protocol specification used by VSCode, Atom and other popular editors/IDEs.
+Most help and completion issues lie outside of Spyder's control, and are either limitations with PyLS or the code that is being introspected, but some can be worked around.
+
+
+Object missing docstring
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+If nothing is displayed in the calltip, hover hint or help pane, the object you're trying to introspect may not have a docstring.
+
+.. image:: images/common-illnesses/common-illnesses-missing-docstring.png
+   :alt: Docstring not found in help pane
+   :width: 500px
+
+In this case, the only solution is to add one in the source code of the original function, method or class.
+
+
+Object cannot be found
+~~~~~~~~~~~~~~~~~~~~~~
+
+Some objects, whether due to being written in C, Cython or another language; generated dynamically at runtime; or being a method of an object you create, cannot be easily found without executing the code.
+
+.. image:: images/common-illnesses/common-illnesses-not-found.png
+   :alt: Object not found in help pane
+   :width: 500px
+
+However, once you run your code in the :doc:`ipythonconsole`, you might be able to get help and completions on the object there.
+
+
+LSP has stopped working
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Occasionally, especially after using Spyder for a while, code completion, help and analysis may stop working.
+If this is the case, you can check LSP status with the :guilabel:`LSP Python` item in Spyder's status bar at the bottom of the screen, and restart it by right-clicking it and selecting the :guilabel:`Restart Python Language Server` item.
+
+.. image:: images/common-illnesses/common-illnesses-LSP-restart.png
+   :alt: Spyder with LSP restart dialog
+
+
+Spyder bug/dependency issue
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Given the variety of dependencies involved in making LSP work, an incompatible or out of date version in your environment can result in error messages, incomplete results, or help/analysis not working at all.
+
+To address this, first try updating Anaconda and Spyder as described in :doc:`basic-first-aid`.
+If the issue still isn't resolved, update the various relevant dependencies with:
+
+.. code-block:: bash
+
+   conda update python-language-server
+
+
+
+===============
+Plugin Problems
+===============
+
+If the error mentions or involves a Spyder plug-in, such as ``spyder-unittest``, ``spyder-terminal`` or ``spyder-notebook``, the first approach should be to update Spyder and the plugin to their latest versions.
+
+If this doesn't fix the problem, you should check the plugin's website or repository to see if it is compatible with your version of Spyder.
+
+Finally, if compatibility doesn't seem to be the problem, please check those repositories to see if an issue was already opened, and report it there if not.
```

Comment:
Here is an added section:

```suggestion

Plugin does not work at all
~~~~~~~~~~~~~~~~~~~~~~~~~~~
If you have installed a Spyder plug-in, but Spyder does not seem to see it, then first select the :guilabel:`Dependencies` item under the :guilabel:`Help` menu. All Spyder plug-ins are listed at the bottom of the list of dependencies.

If the plug-in with the problem is listed in the list of dependencies, then go to the :guilabel:`Panes` submenu of the :guilabel:`View` menu and select the plug-in. This makes the plug-in pane visible.

If the plug-in with the problem is not listed in the list of dependencies, then check that you installed it in the same environment as Spyder. If you have, then the problem may well be caused by a dependency issue. Test whether you can import the plug-in manually by opening a Python console in the same environment as Spyder and typing for instance `import spyder_unittest` to test the `spyder-unittest` plug-in; this command should run without errors.

If none of this helps you to resolve the problem, then continue as in the next section.

Other issues
~~~~~~~~~~~~
If you get an error which mentions or involves a Spyder plug-in, such as ``spyder-unittest``, ``spyder-terminal`` or ``spyder-notebook``, or if you encounter any other problem with a Spyder plug-in, then the first approach should be to update Spyder and the plugin to their latest versions.

If this doesn't fix the problem, you should check the plugin's website or repository to see if it is compatible with your version of Spyder.

Finally, if compatibility doesn't seem to be the problem, please check those repositories to see if an issue was already opened, and report it there if not.
```

Reply:
Thank you very much @jitseniesen. I'm committing your suggestion so @CAM-Gerlach can take a final review to it just to make sure format and style are fine.